CMakeLists.txt

```diff
@@ -318,18 +320,20 @@ add_custom_command (OUTPUT ${DIST_FILE}
       ${ALL_DIST} ${dist_generated_depends}
 )
 
-add_custom_target (dist DEPENDS ${DIST_FILE})
-
-add_custom_target (distcheck DEPENDS dist
-   COMMAND ${CMAKE_COMMAND}
-      -D CMAKE_MODULE_PATH=${PROJECT_SOURCE_DIR}/cmake/make_dist
-      -D CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
-      -D PACKAGE_PREFIX=${PACKAGE_PREFIX}
-      -D CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-      -P ${PROJECT_SOURCE_DIR}/cmake/make_dist/MakeDistCheck.cmake
-)
+if(NOT (TARGET bson_shared OR TARGET mongoc_shared))
+   add_custom_target (dist DEPENDS ${DIST_FILE})
+
+   add_custom_target (distcheck DEPENDS dist
+      COMMAND ${CMAKE_COMMAND}
+         -D CMAKE_MODULE_PATH=${PROJECT_SOURCE_DIR}/cmake/make_dist
+         -D CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
+         -D PACKAGE_PREFIX=${PACKAGE_PREFIX}
+         -D CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+         -P ${PROJECT_SOURCE_DIR}/cmake/make_dist/MakeDistCheck.cmake
+   )
+endif()
 
-if(ENABLE_UNINSTALL)
+if(ENABLE_UNINSTALL AND NOT (TARGET bson_shared OR TARGET mongoc_shared))
```

Comment:
Similarly, should this take the value of `BUNDLED_MONGOC` into account?

Reply:
I think it should not be taken into account for the same reasons I described above.  If my assumption there is wrong, then the same approach taken for `dist` and `distcheck` should be taken here.

Reply:
Sounds good to me.